pkg/logger/worker.go

```diff
@@ -167,11 +236,24 @@ func (w *Worker) Start() {
 
 			select {
 			case work := <-w.Work:
-				// Receive a work request.
-				w.Log.Infof("Received work request %d, url: %s, requestId: %s", w.ID, work.Url.String(), work.Id)
+				if len(work) == 1 {
+					workItem := work[0]
+					// Receive a work request.
+					w.Log.Infof("Received work request %d, url: %s, requestId: %s", w.ID, workItem.Url.String(), workItem.Id)
+
+					if err := w.sendCloudEvent(workItem); err != nil {
+						w.Log.Error(err, "Failed to send cloud event, url: %s", workItem.Url.String())
+					}
+				} else if len(work) == 2 {
+					// Receive a work request.
+					w.Log.Infof("Received work request %d, url: %s, requestIds: %s %s", w.ID, work[0].Url.String(), work[0].Id, work[1].Id)
+
+					if err := w.sendBatchCloudEvent(work); err != nil {
+						w.Log.Error(err, "Failed to send cloud event, url: %s", work[0].Url.String())
+					}
 
-				if err := w.sendCloudEvent(work); err != nil {
-					w.Log.Error(err, "Failed to send cloud event, url: %s", work.Url.String())
+				} else {
```

Comment:
Is there a possible case with 0 work items ?

Reply:
Right now we control access to the queue through the functions `sendCloudEvent` and `sendBatchCloudEvent`, but I agree we should be defensive on this. I'll make this error log more inclusive to account for the 0 case.